clients/haskell/hs-cac-client/src/Client.hs

```diff
@@ -80,12 +81,22 @@ getError = c_last_error_message
 cleanup :: [Ptr a] -> IO ()
 cleanup items = mapM free items $> ()
 
-createCacClient:: Tenant -> Integer -> String -> IO (Either Error ())
-createCacClient tenant frequency hostname = do
+allocateAndPoke :: Maybe Integer -> IO (Ptr CULong)
```

Comment:
should we name it something like
```suggestion
allocateCLongPtr :: Maybe Integer -> IO (Ptr CULong)
```
although its using `poke` internally, having `poke` in name, made it really confusing to understand what was going on here